APTITUDE/APTITUDETEST.Data/Data/TestRepository.cs

```diff
@@ -165,6 +165,126 @@ public async Task<JsonResult> UpdateTestGroup(UpdateTestGroupVM updateTest)
                 });
             }
 
+        }
+
+        public async Task<JsonResult> AddTestQuestions(AddTestQuestionsVM addTestQuestion)
+        {
+            try
+            {
+                Test test = await Task.FromResult(_context.Tests.Where(t => t.Id == addTestQuestion.TestId && t.Status == (int)Common.Enums.TestStatus.Active && t.IsDeleted == false).FirstOrDefault());
+                if (test == null)
+                {
+                    return new JsonResult(new ApiResponse<string>
+                    {
+                        Message = string.Format(ResponseMessages.NotFound, ModuleNames.Test),
+                        Result = true,
+                        StatusCode = ResponseStatusCode.OK
+                    });
+                }
+
+                #region Code to check questions are available in DB
+                Func<TestQuestionsCountVM, int> func = x => x.OneMarkQuestion;
+                var MarkQuestionCountReq = 0;
+                for (int questionType = 1; questionType <= 2; questionType++)
+                {
+                    for (int i = 1; i <= 5; i++)
+                    {
+                        int MarkQuestionCountInDB = await Task.FromResult(_context.Questions.Where(t => t.Topic == addTestQuestion.TopicId && t.QuestionType == questionType && t.Difficulty == i && t.IsDeleted == false).Count());
```

Comment:
Get questions from db outside the loop, store in local variable and here get specific record from local variable

Reply:
Okay